scripts/test/Muon/basic_fitting_context_test.py

```diff
@@ -106,6 +97,67 @@ def test_that_guess_workspace_name_will_not_raise_if_the_name_provided_is_none(s
         self.fitting_context.guess_workspace_name = None
         self.assertEqual(self.fitting_context.guess_workspace_name, None)
 
+    def test_that_clear_will_clear_the_undo_data_and_active_fits_list(self):
+        self.fitting_context.active_fit_history = [mock.Mock(), mock.Mock()]
+        self.fitting_context.dataset_indices_for_undo = [0, 1]
+        self.fitting_context.single_fit_functions_for_undo = [mock.Mock(), mock.Mock()]
+        self.fitting_context.fit_statuses_for_undo = ["Success", "Fail"]
+        self.fitting_context.chi_squared_for_undo = [2.2, 3.3]
+
+        self.fitting_context.clear()
+
+        self.assertEqual(self.fitting_context.active_fit_history, [])
+        self.assertEqual(self.fitting_context.all_latest_fits(), [])
+        self.assertEqual(self.fitting_context.dataset_indices_for_undo, [])
+        self.assertEqual(self.fitting_context.single_fit_functions_for_undo, [])
+        self.assertEqual(self.fitting_context.fit_statuses_for_undo, [])
+        self.assertEqual(self.fitting_context.chi_squared_for_undo, [])
+
+    def test_that_all_latest_fits_will_return_the_two_most_recent_unique_fits(self):
+        output_ws = WorkspaceFactory.create("Workspace2D", NVectors=3, YLength=5, XLength=5)
+        table_workspace = WorkspaceFactory.createTable()
+
+        output_ws_wrap1 = StaticWorkspaceWrapper("Output1", output_ws)
+        parameter_ws_wrap1 = StaticWorkspaceWrapper("Parameter1", table_workspace)
+        covariance_ws_wrap1 = StaticWorkspaceWrapper("Covariance1", table_workspace)
+        output_ws_wrap2 = StaticWorkspaceWrapper("Output2", output_ws)
+        parameter_ws_wrap2 = StaticWorkspaceWrapper("Parameter2", table_workspace)
+        covariance_ws_wrap2 = StaticWorkspaceWrapper("Covariance2", table_workspace)
+
+        fit1 = FitInformation(["Input1"], "GausOsc", [output_ws_wrap1], parameter_ws_wrap1, covariance_ws_wrap1)
+        fit2 = FitInformation(["Input2"], "GausOsc", [output_ws_wrap2], parameter_ws_wrap2, covariance_ws_wrap2)
+        fit3 = FitInformation(["Input1"], "GausOsc", [output_ws_wrap1], parameter_ws_wrap1, covariance_ws_wrap1)
+
+        self.fitting_context.active_fit_history = [fit1, fit2, fit3]
+
+        self.assertEqual(self.fitting_context.active_fit_history[0], fit1)
+        self.assertEqual(self.fitting_context.active_fit_history[1], fit2)
+        self.assertEqual(self.fitting_context.active_fit_history[2], fit3)
+
+        self.assertEqual(self.fitting_context.all_latest_fits()[0], fit2)
```

Comment:
why is fit1 not at index 0?

Reply:
Because `fit3` has the same input workspaces/output workspaces as `fit1`. `fit3` happened more recently than `fit1` as it is on the end of the fit history list. And `all_latest_fits` returns the most recent 'unique' fits

I'll add a comment to try make this clearer